examples/tutorials/plot_5_warm_starting.py

```diff
@@ -0,0 +1,375 @@
+        test_loss = test_epoch(model, device, test_loader)
+        scheduler.step()
+
+    if args.save_model:
+        run_working_dir = Path(os.environ.get("ORION_WORKING_DIR", "."))
+        # use the trial working dir to save the model.
+        torch.save(model.state_dict(), str(run_working_dir / "model.pt"))
+    return [dict(name="loss", type="objective", value=test_loss)]
+
+
+# %%
+# Controls for this example:
+#
+previous_experiment_n_runs = 10
+previous_experiment_settings = {
+    "dataset": "CIFAR100",
+    "epochs": 3,
+}
+
+current_experiment_n_runs = 10
+current_experiment_settings = {
+    "dataset": "CIFAR10",
+    "epochs": 3,
+}
+
+# We're using multiple seeds for a more robust comparison of with/without warm-starting.
+n_seeds = 3
+
+# The number of initial random suggestions that the optimization algorithm should do.
+n_initial_random_suggestions = 5
+
+# NOTE: This gets run in the tutorials directory
+# NOTE: This needs to be a relative path, otherwise the CI runs will fail.
+# Specify the database where the previous experiments are stored. We use a local PickleDB here.
+previous_experiment_storage = {
+    "type": "legacy",
+    "database": {
+        "type": "pickleddb",
+        "host": "previous_db.pkl",
+    },
+}
+current_experiment_storage = {
+    "type": "legacy",
+    "database": {
+        "type": "pickleddb",
+        "host": "current_db.pkl",
+    },
+}
+
+
+previous_experiment = build_experiment(
+    name="previous_experiment",
+    space={"lr": "loguniform(1e-5, 1.0)"},
+    storage=previous_experiment_storage,
+    algorithms={"random": {"seed": 1}},
+    max_trials=previous_experiment_n_runs,
+    executor=SingleExecutor(),
+)
+
+# %%
+# Populate the initial experiment with some trials:
+
+previous_experiment.workon(main, **previous_experiment_settings)
+
+# %%
+# Run a new experiment, without warm-starting (a.k.a. "cold-start"):
+
+
+cold_experiments = [
+    build_experiment(
+        name=f"cold_experiment_{seed}",
+        space={"lr": "loguniform(1e-5, 1.0)"},
+        storage=current_experiment_storage,
+        executor=SingleExecutor(),
+        algorithms={
+            "tpe": {"seed": seed, "n_initial_points": n_initial_random_suggestions}
+        },
+        # algorithms={"robo_gp": {"seed": seed, "n_initial_points": n_initial_points}},
+        max_trials=current_experiment_n_runs,
+    )
+    for seed in range(n_seeds)
+]
+for exp in cold_experiments:
+    exp.workon(main, **previous_experiment_settings)
+
+#%%
+# New experiment with warm-starting:
+
+assert previous_experiment.storage
+assert previous_experiment.max_trials
+warm_experiments = [
+    build_experiment(
+        name=f"warm_experiment_{seed}",
+        space={"lr": "loguniform(1e-5, 1.0)"},
+        storage=current_experiment_storage,
+        executor=SingleExecutor(),
+        max_trials=current_experiment_n_runs,
+        # NOTE: This n_initial_points is changed slightly, since it also counts the trials from
```

Comment:
It's interesting to make this fair comparison. In practice however would not need to force random points first at the previous results already provide a good basis to start bayesian optimization.

Reply:
Ok, should I keep this then? or not?